mesonbuild/backend/backends.py

```diff
@@ -398,36 +402,43 @@ def object_filename_from_source(self, target, source, is_unity):
                                          os.path.join(self.environment.get_source_dir(), target.get_subdir()))
         return source.replace('/', '_').replace('\\', '_') + '.' + self.environment.get_object_suffix()
 
-    def determine_ext_objs(self, target, extobj, proj_dir_to_build_root):
+    def determine_ext_objs(self, extobj, proj_dir_to_build_root):
         result = []
+
+        # Merge sources and generated sources
+        sources = list(extobj.srclist)
+        for gensrc in extobj.genlist:
+            for s in gensrc.get_outputs():
+                path = self.get_target_generated_dir(extobj.target, gensrc, s)
+                dirpart, fnamepart = os.path.split(path)
+                sources.append(File(True, dirpart, fnamepart))
+
+        # Filter out headers and all non-source files
+        sources = [s for s in sources if self.environment.is_source(s) and \
+                                         not self.environment.is_header(s)]
```

Comment:
__[Flake8]__

[E127] continuation line over-indented for visual indent


<!-- Comment by [SideCI](https://sideci.com) -->